src/OpenTelemetry/Metrics/Metric.cs

```diff
@@ -97,6 +98,11 @@ public sealed class Metric
                 {
                     aggType = AggregationType.HistogramSumCount;
                 }
+                else if (histogramRecordMinMax)
```

Comment:
MinMax should be independent of bounds.
You'll need some logic like 
Start with MetricType.Histogram
If RecordMinMax, set it to HistogramWithMinMax

For aggregationtype (its internal usage only):
We'll likely need:
Histogram  -- count,sum,buckets
HistogramSumCount - count,sum
HistogramMinMax - count, sum, minmax, buckets
HistogramSumCountMinMax - count,sum,minmax

(would be good to rename the internal Aggtype to better reflect intent)